src/auto-complete.jsx

```diff
@@ -343,6 +349,10 @@ const AutoComplete = React.createClass({
       }
     });
 
+    if (this.props.menuItemsLimit > 0 && requestsList.length > this.props.menuItemsLimit) {
```

Comment:
https://github.com/vishalvijay/material-ui/blob/235459e21e3b63f9e83b061a0335094812d4da07/src/auto-complete.jsx#L358-L378

Please have a look at L375 (in above link)

I think its better to do the logic inside the loop (change `map` to `for` loop) because if I need to show only 3 items, the loop doesn't need to prepare all other filtered menu items